Cache/FileCache.php

```diff
@@ -29,19 +29,18 @@ class FileCache implements CacheInterface, CacheWarmerInterface, CacheClearerInt
     public function __construct($baseDir, $dir, $umask = 0002)
     {
         if (!is_dir($baseDir)) {
-            throw new \InvalidArgumentException(sprintf('The directory "%s" does not exist.', $dir));
+            throw new \InvalidArgumentException(sprintf('The directory "%s" does not exist.', $baseDir));
         }
         if (!is_writable($baseDir)) {
-            throw new \InvalidArgumentException(sprintf('The directory "%s" is not writable.', $dir));
+            throw new \InvalidArgumentException(sprintf('The directory "%s" is not writable.', $baseDir));
         }
 
-        $this->dir = rtrim($dir, '\\/');
+        $this->dir = $baseDir.DIRECTORY_SEPARATOR.rtrim($dir, '\\/');
 
         $this->umask = $umask;
 
-        $dir = $baseDir.DIRECTORY_SEPARATOR.$dir;
-        if (!is_dir($dir)) {
-            mkdir($dir, 0777 & ~$this->umask, true);
+        if (!is_dir($this->dir)) {
+            mkdir($this->dir, 0777 & ~$this->umask, true);
```

Comment:
@lsmith77 btw: is good to have 777 for the file cache stuff? thought 775 for app/cache etc is enough?